dapper/mods/integration.py

```diff
@@ -12,8 +12,13 @@
 
 
 # fmt: off
-def rk4(f, x, t, dt, order=4):
-    """Runge-Kutta (explicit, non-adaptive) numerical ODE solvers.
+def rk4(f, x, t, dt, stages=4, s=0.0):
+    """Runge-Kutta (explicit, non-adaptive) numerical (S)ODE solvers.
+
+    The rule has strong / weak convergence order 1.0 for generic SDEs and order 4.0
+    convergence for ODEs when stages=4.  For stages=1, this becomes the Euler-Maruyama
+    schemefor SDEs (s > 0.0) with strong / weak convergence order 1.0 for SDEs with
+    additive noise as defined in the below. See `bib.grudzien2020numerical`.
```

Comment:
Actually regarding this -- how do we set options so that line 244 of  DAPPER/dapper/mods/__init__.py

```
xx[k] = Dyn(xx[k-1], t-dt, dt) + np.sqrt(dt)*Dyn.noise.sample(1)
```
doesn't add additional noise?  I think this may be the only compatibility issue, is the noise a function that we have to supply in the Dyn dict?  Should I just set a lambda function to return zero in general?

Otherwise this model configuration can be used with multiplicative or additive inflation / Gaussian mixtures / etc... to handle issues like sampling error or to increase the variance of the estimator.  The example that I am writing will be treated as a perfect-random model configuration in which the truth twin and each ensemble member will be propagated by the same random model under different outcomes.  Note, the observations are still discrete-in-time so that this is not equivalent to the Kalman-Bucy filter, but rather represents the case in which the truth twin and ensemble members are actually evolved with respect to statistically equivalent, random diffeomorphisms.  In this case, many of the usual issues of sampling error due to nonlinearity etc. still apply, though the SDE evolution tends to regularize some stochastic estimators like particle filters which are more degenerate in totally deterministic systems.  This also tends to regularize some of the issues with the EnKF's need to use random rotations, etc. to numerically regularize the collapse of variances.

Reply:
Using `HMM = modelling.HiddenMarkovModel(Dyn={..., 'noise': 0}, ...)` should do it. But that means that your diffusion/noise must be specified elsewhere. Presumably, for perfect-yet-stochastic-model-case, it should just be inherent to the step function. I believe, for now at least, that is the easiest solution.

Reply:
That makes sense to me -- I'll get a draft of this running shortly.